### PR TITLE
Add scroll-down on showcase-site when filters are applied

### PR DESCRIPTION
**Pull request by MikeBudimon**
Add the feature that any interactions with filters cause the Site Showcase to scroll down past the "Featured" header to the beginning of the filtered Showcase area.
Issue: #9048

### CODE REVIEW

File: www/package.json [hunk 1]
```diff
@@ -67,6 +67,7 @@
     "react-icons": "^2.2.7",
     "react-instantsearch": "^4.5.1",
     "react-modal": "^3.4.4",
+    "react-scroll": "^1.7.10",
```
Comment (pieh) [inline www/package.json]: https://bundlephobia.com/result?p=react-scroll@1.7.10 this adds 6kB to bundle, I don't think it's worth extra bandwitdh for scrolling animation (which we aren't even using anywhere else)

**Comment by KyleAMathews** [comment]
Perhaps this could do it? https://developers.google.com/web/updates/2016/02/smooth-scrolling-in-chrome-49

**Comment by pieh** [comment]
I'm not sure if we want smooth scrolling at all (we don't do that anywhere else). 

Maybe using something like:
```jsx
import { navigate } from "gatsby"

...

navigate(`/showcase/?filters=params#reach-skip-nav`)
```
would be good and quick solution?

File: www/src/views/showcase/index.js [hunk 2]
```diff
@@ -8,6 +8,32 @@ import Layout from "../../components/layout"
...
+      this.scrollToElement(`#showcase`)
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.location.search !== this.props.location.search) {
+      const { location } = this.props
+      if (location.search && location.search.includes(`?filters`)) {
+        this.scrollToElement(`#showcase`)
+      }
```
[the remark this answers is not in the document]
Comment (MikeBudimon) [reply www/src/views/showcase/index.js]: I mean actually it's not worth to DRY it out for only two calls but is it bad? 
Should I just add the scroll code directly into the componentDidMount and componentDidUpdate ?

[the remark this answers is not in the document]
Comment (jlengstorf) [reply www/src/views/showcase/index.js]: I was thinking you could abstract the logic into one place so you can call something like:

```
componentDidMount() {
  this.handleScroll(this.props, `#showcase`);
}
```

But, honestly, I don't know if it's worth the effort. I think what you've got it fine.

[the remark this answers is not in the document]
Comment (pieh) [reply www/src/views/showcase/index.js]: Hey, I did some more browsing around `www` code and it turns out we have https://github.com/gatsbyjs/gatsby/blob/master/www/src/utils/scroll-to-anchor.js scroll util we can use for that.

I also discovered that we actually do have some scrolling animation when clicking "View all" button/link in the "Featured Sites" section that use that uitl (but there is something broken there and scroll goes back - so will need to fix that). But we could just reuse that util here

[the remark this answers is not in the document]
Comment (MikeBudimon) [reply www/src/views/showcase/index.js]: Ok, I will look into it.

[the remark this answers is not in the document]
Comment (pieh) [reply www/src/views/showcase/index.js]: quick update: broken thing I mentioned in previous comment that snaps back scroll is in gatsby runtime not in www, so fixing it is definitely out of scope for this PR.

[the remark this answers is not in the document]
Comment (MikeBudimon) [reply www/src/views/showcase/index.js]: I don't know why but the scroll-to-anchor.js util doesn't work in the showcase/index.js file.
I tried `scrollToAnchor(this.showcase.current, ()=>{})` with other things but it didn't work at all. 
Maybe it has something to do how react mounts refs but then it should work on componentDidUpdate().
Menawhile I pushed new code with working smooth scrolling and offsetTop using `window.scroll({..., behavior: `smooth`,})`

**Comment by wardpeet** [comment]
I've marked this one as blocked. We need to fix #11001 first before going further with this one. Adding scroll to the current behaviour is even more annoying 😄.

onscroll

**Comment by wardpeet** [comment]
Thanks so much for the PR!

It seems like this issue got resolved by #12105 as I did more refactoring than I thought. Sorry

As such--we're going to close this out. We want you to know that we certainly want to keep you as a contributor to Gatsby, so we'd encourage you to check out our open issues.

Thanks again, and we look forward to seeing more PRs from you in the future! 💪 💜

**Pull request closed without merging**